src/__tests__/components/ArticleCreate.test.js

```diff
@@ -0,0 +1,352 @@
+    const newProps = props;
+    newProps.match.params.articleSlug = 'article-slug';
+    const wrapper = mount(<ArticleCreate {...newProps} />);
+    jest.runAllTimers();
+    wrapper.update();
+    expect(wrapper.props().isLoggedIn).toBeTruthy();
+  });
+
+  test('should handle title input', () => {
+    const event = { target: { name: 'title', value: 'hello world' } };
+    const wrapper = mount(<ArticleCreate {...props} />);
+
+    const titleInput = wrapper.find('[data-el="article-input"]');
+    titleInput.simulate('change', event);
+    expect(props.onInputChange).toHaveBeenCalled();
+  });
+
+  test('should handle editor state update', () => {
+    const event = { target: { name: 'title', value: 'hello world' } };
+    const wrapper = mount(<ArticleCreate {...props} />);
+    const tagForm = wrapper.find('[data-test="tag-form"]');
+    const tagInput = wrapper.find('[data-test="tag-input"]');
+    tagInput.simulate('change', event);
+    tagForm.simulate('submit');
+    expect(props.onTagFormSubmit).toHaveBeenCalled();
+  });
+
+  test('should save and publish', () => {
+    const newProps = { ...props, isLoggedIn: true };
+    const wrapper = mount(<ArticleCreate {...newProps} />);
+    wrapper.setState({ isEdit: false });
+    const publishBtn = wrapper.find('[data-el="publish-btn"]');
+    publishBtn.simulate('click');
+    expect(props.postArticle).toHaveBeenCalled();
+  });
+
+  test('should save as drafft', () => {
+    const newProps = { ...props, isLoggedIn: true };
+    const wrapper = mount(<ArticleCreate {...newProps} />);
+    wrapper.setState({ isEdit: false });
+    const publishBtn = wrapper.find('[data-el="draft-btn"]');
+    publishBtn.simulate('click');
+    expect(props.postArticle).toHaveBeenCalled();
+  });
+
+  test('should redirect to auth page', () => {
+    jest.useFakeTimers();
+    const newProps = { ...props, isLoggedIn: false };
+    const wrapper = mount(<ArticleCreate {...newProps} />);
+    jest.runAllTimers();
+    wrapper.update();
+    expect(wrapper.props().isLoggedIn).toBeFalsy();
+  });
+
+  describe('Video handler', () => {
+    const config = [
+      {
+        pattern: API_URL,
+        fixtures: () => ({
+          body: { secure_url: urlValue },
+        }),
+        post: (match, data) => data,
+      },
+    ];
+    const event = {
+      target: {
+        files: [JSON.stringify(file)],
+      },
+    };
+    mock(request, config);
+    test('should open video input', (done) => {
+      const wrapper = mount(<ArticleCreate {...props} />);
+      const prevState = wrapper.state().editorState;
+      const videoBtn = wrapper.find('[data-el="video-btn"]');
+      videoBtn.simulate('click');
+      wrapper.find('[data-el="video-input"]')
+        .simulate('change', event);
+      wrapper.find('[data-el="video-input"]');
+      const newState = wrapper.state().editorState;
+      expect(prevState).toEqual(newState);
+      done();
+    });
+    test('should add video', (done) => {
```

Comment:
please add space between tests! not only here

Reply:
I am on it thanks!!!